src/main/java/com/hazelcast/aws/EcsClientStrategy.java

```diff
@@ -0,0 +1,43 @@
+package com.hazelcast.aws;
+
+import com.hazelcast.aws.impl.DescribeTasks;
+import com.hazelcast.aws.impl.ListTasks;
+
+import java.net.URL;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ *
+ */
+class EcsClientStrategy extends AwsClientStrategy {
+
+    EcsClientStrategy(AwsConfig awsConfig, String endpoint) {
+        super(awsConfig, endpoint);
+    }
+
+    @Override
+    public Collection<String> getPrivateIpAddresses() throws Exception {
+        return getAddresses().keySet();
+    }
+
+    @Override
+    public Map<String, String> getAddresses() throws Exception {
+        // FIXME taskMetadata URL
+        // String clusterName = new TaskMetadata(awsConfig, new URL("https", endpoint, -1, "/")).execute();
+        ListTasks listTasks = new ListTasks(awsConfig, new URL("https", endpoint, -1, "/"));
+        Collection<String> taskArns = listTasks.execute(/*clusterName*/);
+        if (!taskArns.isEmpty()) {
+            DescribeTasks describeTasks = new DescribeTasks(awsConfig, new URL("https", endpoint, -1, "/"));
+//            describeTasks.setSecurityToken(listTasks.getSecurityToken());
+            return describeTasks.execute(taskArns);
+        }
+        return Collections.EMPTY_MAP;
+    }
+
+    @Override
+    public String getAvailabilityZone() {
+        return "ecs";
```

Comment:
I still have to determine if we have a valid AZ concept on Fargate, I think we can implement it on EC2 classic. In the meanwhile a fixed string is required by discovery API.

Reply:
Ok, so after this PR, the ZONE_AWARE feature does not work for Fargate, right? Could you create a GH issue? We should implement it , but it can be a separate PR.

Reply:
opened #98